distributed/security.py

```diff
@@ -153,18 +153,57 @@ def _set_field(self, kwargs, field, config_name):
             out = dask.config.get(config_name)
         setattr(self, field, out)
 
-    def __repr__(self):
+    def attr_to_dict(self):
         keys = sorted(self.__slots__)
         keys.remove("extra_conn_args")
-        items = []
+
+        attr = {}
+
         for k in keys:
             val = getattr(self, k)
             if val is not None:
                 if isinstance(val, str) and "\n" in val:
-                    items.append((k, "..."))
+                    attr[k] = "Temporary (In-memory)"
+                elif isinstance(val, str):
+                    attr[k] = f"Local ({os.path.abspath(val)})"
                 else:
-                    items.append((k, repr(val)))
-        return "Security(" + ", ".join(f"{k}={v}" for k, v in items) + ")"
+                    attr[k] = val
+
+        return attr
+
+    def __repr__(self):
+        attr = self.attr_to_dict()
+        return (
+            "Security("
+            + ", ".join(f"'{key}'='{value}'" for key, value in attr.items())
```

Comment:
Let's remove the single quotes here. Folks may think things are strings when they are not.

```suggestion
            + ", ".join(f"{key}={value}" for key, value in attr.items())
```